### ISSUE TITLE
ingress: support widely-supported ingress annotations

### ISSUE THREAD

**Issue by nikhiljha** [cilium/cilium#22887]
Cilium should support Ingress annotations that other ingress controllers have for better compatibility and usability. I stole the list below from the [Contour docs](https://projectcontour.io/docs/v1.0.0/annotations/). Contour is a good place to look because their Ingress is also backed via Envoy, so there's a working example of the Envoy configuration needed to make these annotations work.

## Particularly Common Annotations

I plan to PR these over the next few weeks / months as I need them 😅 -- if that's something that we want.

- [x] `kubernetes.io/ingress.class` (#22629)
- [ ] `ingress.kubernetes.io/force-ssl-redirect`
- [ ] `kubernetes.io/ingress.allow-http`
- [ ] `io.cilium/tls-minimum-protocol-version` [envoy docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/api-v2/api/v2/auth/cert.proto#envoy-api-msg-auth-tlsparameters)
- [X] `io.cilium/websocket` (#20814)
- [ ] `io.cilium/upstream-protocol.{protocol}` (needed for upstream TLS, which currently isn't allowed by Cilium)

## Advanced/Misc Annotations

- [ ] `io.cilium/num-retries` [envoy docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/configuration/http_filters/router_filter.html#config-http-filters-router-x-envoy-max-retries)
- [ ] `io.cilium/per-try-timeout` [envoy docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/api-v2/api/v2/route/route.proto#envoy-api-field-route-routeaction-retrypolicy-retry-on)
- [ ] `io.cilium/response-timeout` [envoy docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/api-v2/api/v2/route/route.proto.html#envoy-api-field-route-routeaction-timeout)
- [ ] `io.cilium/retry-on` [envoy docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/api-v2/api/v2/route/route.proto#envoy-api-field-route-routeaction-retrypolicy-retry-on), [more docs](https://www.envoyproxy.io/docs/envoy/v1.11.2/configuration/http_filters/router_filter.html#config-http-filters-router-x-envoy-retry-on)

**Comment by swastik959** [comment]
@nikhiljha I have tried to work on this issue please review the pull request and suggest some necessary changes

**Comment by chaunceyjiang** [comment]
Hi @christarazi @nbusseneau @sayboras 

I see that there are still two annotations that have not been implemented. 
`ingress.kubernetes.io/force-ssl-redirect , kubernetes.io/ingress.allow-http`

Do we still need these two annotations now?


if need, i can pick up this task.

**Comment by steled** [comment]
Does it make sense/is possible to also include [Configuration snippet](https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#configuration-snippet)?

**Comment by joestringer** [comment]
I think that at this point we can probably close this issue in favor of using Gateway API for configuring common Ingress functionality. What do you think @cilium/sig-servicemesh ?

**Comment by youngnick** [comment]
@joestringer yes, I think that's probably for the best. We're minimising additions to the Ingress reconciler now that Ingress is frozen.

@steled, We definitely wouldn't add Configuration Snippet support, for two reasons:

* It's for nginx config, and Cilium's Ingress uses Envoy
* Inlining config directly into the proxy config has been the source of _numerous_ CVEs in ingress-nginx, it's _not_ a good idea. Gateway API is specifically designed to allow for more flexibility in config without also allowing arbitrary inlining, with all the security issues that brings.
